### PR TITLE
Add "columns" and "square" to grid card editor

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

Add the `columns` and `square` options to graphical card editor.

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by iantrich** [comment]
If I remember correctly, those were left out on purpose

**Comment by spacegaier** [comment]
Really 👀 ? Do you know the reason?

**Comment by zsarnett** [comment]
I don't think they were left out on purpose? My plan was to do something like this eventually having columns and square enabled in the editor. this is the first step
image

**Comment by iantrich** [comment]
If that's the case, then I see no issue in adding this

File: src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts [hunk 1]
```diff
@@ -77,6 +79,35 @@ export class HuiStackCardEditor extends LitElement
 
     return html`
       <div class="card-config">
+        ${this._config.type === "grid"
```
Comment (bramkragten) [inline src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts:82]: We should create a new editor for the grid card, we can still use the logic of the stack card editors, you could extend it.

**Comment by spacegaier** [comment]
@bramkragten As discussed I now de-duplicated the rendering code by re-using the render() from the stack editor and injecting the top section for the grid-card specific elements.

File: src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts [hunk 2]
```diff
@@ -77,6 +69,7 @@ export class HuiStackCardEditor extends LitElement
 
     return html`
       <div class="card-config">
```
Comment (bramkragten) [inline src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts:71]: Why can't we just put the extra elements above of this div?

**Review by bramkragten — APPROVED** [review]

**Pull request merged**